packages/non-core/coffeescript-compiler/coffeescript-compiler.js

```diff
@@ -0,0 +1,214 @@
+import { BabelCompiler } from 'meteor/babel-compiler';
+import CoffeeScript from 'coffeescript';
+import { SourceMapConsumer, SourceMapGenerator } from 'source-map';
+
+
+// The CoffeeScript compiler overrides Error.prepareStackTrace, mostly for the
+// use of coffee.run which we don't use. This conflicts with the tool's use of
+// Error.prepareStackTrace to properly show error messages in linked code.
+// Restore the tool's one after CoffeeScript clobbers it at import time.
+if (Error.METEOR_prepareStackTrace) {
+  Error.prepareStackTrace = Error.METEOR_prepareStackTrace;
+}
+
+
+// The CompileResult for this CachingCompiler is a {source, sourceMap} object.
+CoffeeScriptCompiler = class CoffeeScriptCompiler {
```

Comment:
If you use `api.mainModule` in `package.js`, then this line can just be
```js
export class CoffeeScriptCompiler {
```
and `api.export("CoffeeScriptCompiler", "server")` will use the exported `CoffeeScriptCompiler`.